### PR TITLE
New Charge Ammo and mechanoid ammo tweaks

### PR DESCRIPTION
**Pull request by Tostov**
## Additions

Added the following charge calibers:
10x18 Charged - Charged magnum round
6x22 Charged - Intermediate mechanoid charge round
8x50 Charged - High power charge round
12x65 Charged - Manmade High Caliber charge round

Added a recipe to break down mechanoid charge rounds into a small amount of steel and plasteel (200 rounds into 2 plasteel/steel).

## Changes

Increased the penetration of all current non-mechanoid charge ammo.
(6x18,6x24,8x35)
Tweaked 80x256 Fuel cells cook-off detonation force. They should no longer explode so violently that they can chain detonate other unharmed centipedes.

Reduced centipede magazine count from 36 to 10-12.
Mech charge rounds now all cost the same, and the value has been significantly reduced (0.20 each, this should make centipede ammo costs consistent across the two types).

Overhauled charged shotgun rounds. Scatter and Ion are now both 6, generally stronger pellets. Slug power increased and secondary damage reduced.

## References
Data Sheet
https://docs.google.com/spreadsheets/d/13MUWjDc9UXBEit_pPoHOhLp0xJWk9qE3dRqoqEevGrc/edit?usp=sharing

## Reasoning

Charge ammuntion is currently fairly expensive in terms of time and resource investment per round, and advanced ammo has proven too effective in closing the gap between conventional firearms. By increasing the penetration we can improve their performance against mid to lategame targets without making them excessively lethal against other targets, leaving enough space for advanced ammo to still be viable to a degree, especially for their cost.

There are currently a very small number of avaliable charge calibers avaliable, which limits the potential options when patching more diverse charge weapons. A number of mods currently make use of the mech specific 12x64 round, and by implementing these new calibers we can help make a more concrete divide between the mech made and manmade calibers, allowing us to more freely match the cost of the charge rounds to their specs.

80x256 incendiary fuel cells are currently excessively deadly when detonated as a stack, well capable of destroying nearby centipedes even with reduced ammo storage amounts. By decreasing the cook off potential, it prevents them from being a massive liability and helps fix large centipede blobs being instantly chain detonated to death. In general, centipedes currently carry an excessive amount of ammo, so much that even with reduction each given centipede provides a significant amount of addtional wealth to the player.

## Alternatives

We could buff some other characteristic of charge weapons or just leave them be in their current state. We could signifcicantly nerf advanced ammo.

For centipedes and mechanoid ammo, we could implement a new system that works differentely for them, but I believe even if we were to do this, the changes here would make an acceptable stopgap solution.

## Testing

Check tests you have performed:
- [x] Compiles without warnings
- [x] Game runs without errors
- [x] Playtested a colony (two with variations of this change, somewhere in the range of 10-20 hours in total afaik)

### CODE REVIEW

File: Defs/Ammo/Advanced/80x256mmFuelCell.xml [hunk 1]
```diff
@@ -49,9 +49,10 @@
 	<detonateProjectile>Bullet_80x256mmFuel_Thermobaric</detonateProjectile>
 	<comps>
     <li Class="CompProperties_Explosive">
-        <explosiveRadius>2</explosiveRadius>
+        <explosiveRadius>1</explosiveRadius>
+		<damageAmountBase>20</damageAmountBase>
         <explosiveDamageType>Thermobaric</explosiveDamageType>
-		<explosiveExpandPerStackcount>0.15</explosiveExpandPerStackcount>
+		<explosiveExpandPerStackcount>0.10</explosiveExpandPerStackcount>
```
Comment (Alicecomma) [inline Defs/Ammo/Advanced/80x256mmFuelCell.xml:55]: Just wondering if this is the line preventing chain reactions?

**Review by Alicecomma — APPROVED** [review]

File: Defs/Ammo/Advanced/80x256mmFuelCell.xml (hunk 1 above)
Comment (Tostov) [reply Defs/Ammo/Advanced/80x256mmFuelCell.xml:55]: This or adding the base damage amount, pretty sure.

**Comment by Breadbox2k19** [comment]
May I propose a lower penetration value increase but higher damage value? Charge weaponry being terrible is more due to regular special ammunition being too good and too widely available.

**Comment by Tostov** [comment]
Lowering the penetration and increasing the damage reduces the effective target profiles of charge munitions since they are heavily reliant on sharp for all their existing calibers to inflict damage, makes it better at killing low armor targets which I believe is undesirable from a gameplay curve point of view, and requires that some calibers that don't really need buffing like the HCB also get buffed unless we screw around with giving those specificially different rules.

Because of this, I believe that raising the damage and lowering the penetration is the exact wrong way to combat the problem.

While advanced rounds are certainly potent in their own manner, it is not the only factor in the current lackluster nature of charge rounds, and thus their massive lack of desirability given their significant cost compared to many other munition types. By raising their penetration we can move them away from the conventional rounds and improve their efficency without needlessly increasing the already high lethality of low to mid armor targets. Increased penetration means increased consistency in damage, without increasing the overall damage by too much, since in order to further increase damage much past the initial jump you need massively more penetration.

**Comment by Breadbox2k19** [comment]
> Lowering the penetration and increasing the damage reduces the effective target profiles of charge munitions since they are heavily reliant on sharp for all their existing calibers to inflict damage, makes it better at killing low armor targets which I believe is undesirable from a gameplay curve point of view, and requires that some calibers that don't really need buffing like the HCB also get buffed unless we screw around with giving those specificially different rules.
> 
> Because of this, I believe that raising the damage and lowering the penetration is the exact wrong way to combat the problem.
> 
> While advanced rounds are certainly potent in their own manner, it is not the only factor in the current lackluster nature of charge rounds, and thus their massive lack of desirability given their significant cost compared to many other munition types. By raising their penetration we can move them away from the conventional rounds and improve their efficency without needlessly increasing the already high lethality of low to mid armor targets. Increased penetration means increased consistency in damage, without increasing the overall damage by too much, since in order to further increase damage much past the initial jump you need massively more penetration.

I meant increasing the penetration by a lower degree(the current increase in the PR is a 60% jump), not actually decreasing penetration, I'm not guilty of suggesting such a stupid idea.
The stopping of charge round is simply just lackluster in many circumstances, even lower than many regular rifle caliber round, especially given how much blunt damage even regular ballistic rounds can do.

**Comment by Tostov** [comment]
Quite frankly, the reason people don't use 6x18 or 6x24 is because they don't offer any abilities, of current, to deal with the heavy armor threats that are faced lategame. Nothing short of an absurd buff to damage will make them any more desirable because at the moment they do no damage if they don't penetrate against comparable targets anyway.

As of current state, 6x18 is only able to penetrate recon armor with headshots against the helmet, or after the armor has been sufficently worn down. Likewise it cannot defeat flak + composite until, once again, the armor has been worn down or against a composite helmet with headshots. Increasing the damage makes it better at something it is already more than sufficently good at, while offering no improvement in roles that would actually make the weapon desirable. Part of 6x18's massive jump is also a change in interpretation from regular 45 acp charge to one of the more pdwy calibers like 5.7, 4.6 or 7.62 tokerev. In general there's not any point assuming people in the future will make a gun with no real effective role, and given that drop pods are a thing, it's entirely reasonble to assume that you'd want to give your rear echelon staff something that can actually deal damage rather than tickling people in power armor who come crashing through the roof.

Likewise, 6x24 and 8x35 going up to 15mm pen makes them viable to use against normal quality cataphract armor with conc. rounds without doing literally nothing to it, while improving damage consistentcy against targets with higher quality power armor setups, and moving both out of the critical "super low damage" zone that 8x35 ion rounds currently sit in against centipedes to a degree.

File: Defs/RecipeDefs/Recipes_Production.xml [hunk 2]
```diff
@@ -70,5 +70,43 @@
 		</products>
 	</RecipeDef>
 
+	<RecipeDef>
+		<defName>ReclaimMechAmmo</defName>
```
Comment (N7Huntsman) [inline Defs/RecipeDefs/Recipes_Production.xml:74]: Honestly, I might include a bunch of different varieties of charge ammo. Additionally, you could add a more efficient version for the fabrication bench.

**Pull request merged**